documentation/examples/custom-sd/adapter-usage/main.go

```diff
@@ -0,0 +1,291 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	consul "github.com/hashicorp/consul/api"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/documentation/examples/custom-sd/adapter"
+	"github.com/prometheus/prometheus/util/strutil"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	a             = kingpin.New("sd adapter usage", "Tool to generate file_sd target files for unimplemented SD mechanisms.")
+	outputFile    = a.Flag("output.file", "Output file for file_sd compatible file.").Default("custom_sd.json").String()
+	listenAddress = a.Flag("listen.address", "The address the HTTP sd is listening on for requests.").Default("localhost:8080").String()
+	logger        log.Logger
+
+	// serviceLabel is the name of the label containing the service name.
+	serviceLabel = model.LabelName("consul_service")
```

Comment:
All labels should be prefixed with __meta

Reply:
I removed the `__meta` prefix since the labels will just end up in the file_sd config and then (I assume) be prepended with `__meta` again? If that's not the case I'll add them back with a comment.

Reply:
No, the labels here are the labels. For a non-generic SD you wouldn't want to have to add extra relabelling.